website/0.8.3/src/site/markdown/quota_scheduling.md

```diff
@@ -0,0 +1,181 @@
+
+This new component serves as an iterator for JobDAGs for the Controller. Previously, task assignment required the Controller to iterate through all jobs and their underlying tasks to determine whether there were any tasks that needed to be assigned and scheduled. This proved to be inefficient and did not scale with the increasing load we were putting on Task Framework. Each RuntimeJobDag records states, that is, it knows what task needs to be offered up to the Controller for scheduling. This saves the redundant computation for the Controller every time it goes through the TaskSchedulingStage of the Task pipeline.
+
+![Architecture](/website/0.8.3/src/site/resources/images/quota_InstanceCapacityManager.jpeg)
+
+## User Manual
+
+### Review of Task Framework Glossary
+
+-   Task Framework: a component of Apache Helix. A framework on which users can define and run workflows, jobs, and tasks in a distributed way.
+-   Workflow: the largest unit of work in Task Framework. A workflow consists of one or more jobs. There are two types of workflows:
+    -   Generic workflow: a generic workflow is a workflow consisting of jobs (a job DAG) that are used for general purposes. A generic workflow may be removed if expired or timed out.
+    -   Job queue: a job queue is a special type of workflow consisting of jobs that tend to have a linear dependency (this dependency is configurable, however). There is no expiration for job queues - it lives on until it is deleted.
+-   Job: the second largest unit of work in Task Framework. A job consists of one or more mutually independent tasks. There are two types of jobs:
+    -   Generic job: a generic job is a job consisting of one or more tasks.
+    -   Targeted job: a targeted job differs from generic jobs in that these jobs must have a  _target resource_, and the tasks belonging to such jobs will be scheduled alongside the partitions of the target resource. To illustrate, an Espresso user of Task Framework may wish to schedule a backup job on one of their DBs called  _MemberDataDB_. This DB will be divided into multiple partitions (_MemberDataDB_1, _MemberDataDB_2, ... _MemberDataDB_N)___, and suppose that a targeted job is submitted such that its tasks will be paired up with each of those partitions. This "pairing-up" is necessary because this task is a backup task that needs to be on the same physical machine as those partitions the task is backing up.
+-   Task: the smallest unit of work in Task Framework. A task is an independent unit of work.
+-   Quota resource type: denotes a particular type of resource. Examples would be JVM thread count, memory, CPU resources, etc.. Generally, each task that runs on a Helix Participant (= instance, worker, node) occupies a set amount of resources.  **Note that only JVM thread count is the only quota resource type currently supported by Task Framework, with each task occupying 1 thread out of 40 threads available per Helix Participant (instance).**
```

Comment:
Better to move this section above to have an unified section for all Helix definitions.

Reply:
Combined the two glossary sections into one.